### DISCUSSION TITLE
Is there any plan up upgrade old templates?

### DISCUSSION THREAD

**Discussion by Swampen** [actions/virtual-environments#5988] [General questions] [answered]
Is there any plan to migrate the existing `json` templates e.g. the [`ubuntu2004.json`](https://github.com/actions/virtual-environments/blob/main/images/linux/ubuntu2004.json) template to the new `hcl` packer template format?

If yes, what is the estimated time frame

**Comment by miketimofeev** [comment, marked as answer]
@Swampen yes, we have such a plan but I'm afraid no ETA for now. Please feel free to submit PR if you have a chance to convert the templates.

**Reply by Swampen** [reply]
Thanks for your answer @miketimofeev. I will see if I have some time to convert the ubuntu2004 one. Do you also plan to utilise the plugin functionality in packer as well?

**Reply by miketimofeev** [reply]
Haven't investigated much about it. Could provide what are the benefits of using the plugins?

**Reply by Swampen** [reply]
TL;DR: As of now, Packer comes bundled with support with all sorts of cloud providers. With Packer 2.0 they will remove these from the core binary, and require plugins. The benefit with this is that the Packer binary will be quite small, and a "Packer project" will only install the required plugin (the azure plugin in this case) 
Long version: https://www.hashicorp.com/resources/the-road-to-packer-2-0
